cron/centAcl.php

```diff
@@ -177,20 +185,27 @@ function programExit($msg)
          * Add Hosts
          */
         if ($row['all_hosts']) {
-            $query = "SELECT host_id FROM host WHERE host_id NOT IN (SELECT DISTINCT host_host_id " .
-                "FROM acl_resources_host_relations WHERE acl_res_id = '" . $row['acl_res_id'] . "') " .
-                "AND host_register = '1'";
-            $res1 = $pearDB->query($query);
+            $res1 = $pearDB->prepare(
+                "SELECT host_id FROM host WHERE host_id NOT IN (SELECT DISTINCT host_host_id
+                FROM acl_resources_host_relations WHERE acl_res_id = :aclResId)
+                AND host_register = '1'"
+            );
+            $res1->bindValue(':aclResId', $row['acl_res_id'], \PDO::PARAM_INT);
+            $res1->execute();
 
             if ($res1->rowCount()) {
                 // set acl_resources.changed flag to 1
                 $aclResourcesUpdated = true;
             }
 
             while ($rowData = $res1->fetch()) {
-                $insert_query = "INSERT INTO acl_resources_host_relations (host_host_id, acl_res_id) VALUES ('" .
-                    $rowData['host_id'] . "', '" . $row['acl_res_id'] . "')";
-                $pearDB->query($insert_query);
+                $stmt = $pearDB->prepare(
```

Comment:
Don't forget to create a db transaction when there is more than one update request. (=> while)